drivers/flash/soc_flash_nrf.c

```diff
@@ -488,3 +523,44 @@ static int write(off_t addr, const void *data, size_t len)
 
 	return write_op(&context);
 }
+
+#ifdef CONFIG_NRF52_ANOMALY_242_WORKAROUND
+/* Disable POFWARN by writing POFCON before a write or erase operation.
+ * Do not attempt to write or erase if EVENTS_POFWARN is already asserted.
+ */
+static bool pofcon_enabled;
+static nrf_power_pof_thr_t pofcon_pof_thr;
+
+static int suspend_pofwarn(void)
+{
+	bool enabled;
+	nrf_power_pof_thr_t pof_thr;
+
+	pof_thr = nrf_power_pofcon_get(NRF_POWER, &enabled);
+
+	if (enabled) {
+		nrf_power_pofcon_set(NRF_POWER, false, NRF_POWER_POFTHR_V27);
```

Comment:
Do you need to change the threshold here? Wouldn't it be sufficient to just disable the warning? I mean:
```
		nrf_power_pofcon_set(NRF_POWER, false, pof_thr);
```
You could eliminate `pofcon_pof_thr` then (and use `nrf_power_pofcon_get()` in `restore_pofwarn()`).